drivers/spider.py

```diff
@@ -0,0 +1,114 @@
+"""
+  Spider3 Rider Driver
+"""
+
+import sys
+import os
+import inspect
+
+OSGAR_ROOT = os.path.realpath(os.path.abspath(os.path.join(os.path.split(inspect.getfile( inspect.currentframe() ))[0],"..")))
+if OSGAR_ROOT not in sys.path:
+    sys.path.insert(0, OSGAR_ROOT) # access to logger without installation
+
+
+import serial
+import struct
+from threading import Thread, Event
+
+from lib.logger import LogWriter, LogReader
+
+
+class Spider(Thread):
+    def __init__(self, config, logger, output, name='spider'):
+        Thread.__init__(self)
+        self.setDaemon(True)
+        self.should_run = Event()
+        self.should_run.set()
+
+        self.logger = logger
+        self.stream_id_in = config['stream_id_in']
+        self.stream_id_out = config['stream_id_out']
+
+        self.buf = b''
+        self.output = output
+        self.name = name
+
+        if 'port' in config:
+            self.com = serial.Serial(config['port'], config['speed'], dsrdtr=1)
+            self.com.setRTS()
+            self.com.setDTR(0)
+            self.com.timeout = 0.01  # expects updates < 100Hz
+
+            for i in range(10):
+                data = self.com.read(1024)
+                self.logger.write(self.stream_id_in, data)
+
+            data = b'\xFF'*10
+            self.logger.write(self.stream_id_out, data)
+            self.com.write(data)
+            data = b'\xFE\x57'  # CAN_SPEED_1MB
+            self.logger.write(self.stream_id_out, data)
+            self.com.write(data)
+            data = b'\xFE\x31'  # start bridge
+            self.logger.write(self.stream_id_out, data)
+            self.com.write(data)
+        else:
+            self.com = None
+
+        self.status_word = None  # not defined yet
+
+    @staticmethod
+    def split_buffer(data):
+        # skip 0xFF prefix bytes (CAN bridge control bytes)
+        while len(data) > 0 and data[0] == 0xFF:
+            data = data[1:]
```

Comment:
This is like the most inefficient way to do it 😉. Can we use [lstrip] instead?

```python
data = data.lstrip(b'\xff')
```

[lstrip]: https://docs.python.org/3/library/stdtypes.html#bytearray.lstrip

Reply:
I was aware of that (about the slowness and copy & allocation). This code should be called approximately 8 times on the synchronization at the very beginning ... but `lstrip(b'\xff')` looks better, thanks.